tests/testthat/test-single_quotes_linter.R

```diff
@@ -26,3 +26,10 @@ test_that("returns the correct linting", {
     linter
   )
 })
+
+test_that("handles R>=4.0.0 strings", {
+  skip_if_not_installed("base", "4.0.0")
+  expect_lint('R"(hello \'\' there)"', NULL, single_quotes_linter())
```

Comment:
Just to be sure, `r'(")'` is also a lint, right?

Reply:
you mean not a lint right?

Reply:
No it should lint I think because we only want double quotes unless it requires escapes.

`r"(")"` is fine so no need for `'` at all.

Reply:
Hmm I see what you mean. Not sure I agree though... `r"(")"` may be valid but it's not so readable. `r'(")'` is still clearer IMO

Reply:
I got as far as writing the regex before changing my mind though, I think this works for posterity:

```
rex(
  list(start, single_quote, any_non_double_quotes, single_quote, end) %or%
  list(start, character_class('Rr'), single_quote)
)
```

Reply:
maybe it's best to leave this as an option? if so let's handle in follow-up